src/client/method-status.c

```diff
@@ -282,9 +303,106 @@ static int method_status_unit_on(Client *client, char *node_name, char **units,
+                                        strerror(-r));
+                                return r;
+                        }
+
+                        r = sd_bus_match_signal(
+                                        client->api_bus,
+                                        NULL,
+                                        BC_INTERFACE_BASE_NAME,
+                                        monitor_path,
+                                        MONITOR_INTERFACE,
+                                        "UnitStateChanged",
+                                        on_unit_status_changed,
+                                        user_data);
+                        if (r < 0) {
+                                fprintf(stderr,
+                                        "Failed to create callback for UnitStateChanged: %s\n",
+                                        strerror(-r));
+                                return r;
+                        }
+
+                        _cleanup_sd_bus_message_ sd_bus_message *m = NULL;
+                        r = sd_bus_message_new_method_call(
+                                        client->api_bus,
+                                        &m,
+                                        BC_INTERFACE_BASE_NAME,
+                                        monitor_path,
+                                        MONITOR_INTERFACE,
+                                        "SubscribeList");
+                        if (r < 0) {
+                                fprintf(stderr, "Failed creating subscription call: %s\n", strerror(-r));
+                                return r;
+                        }
+
+                        sd_bus_message_append(m, "s", node_name);
+
+                        r = sd_bus_message_open_container(m, SD_BUS_TYPE_ARRAY, "s");
+                        if (r < 0) {
+                                fprintf(stderr, "Failed opening subscription unit array: %s\n", strerror(-r));
+                                return r;
+                        }
+
+
+                        char *unit_saveptr = NULL;
+                        char *unit = strtok_r(units[i], ",", &unit_saveptr);
+                        for (;;) {
+                                if (unit == NULL) {
+                                        break;
+                                }
+
+                                fprintf(stdout, "Subscribing to node '%s' and unit '%s'\n", node_name, unit);
+                                sd_bus_message_append(m, "s", unit);
+
+                                unit = strtok_r(NULL, ",", &unit_saveptr);
+                        }
+
+                        r = sd_bus_message_close_container(m);
+                        if (r < 0) {
+                                fprintf(stderr, "Failed closing subscription unit array: %s\n", strerror(-r));
+                                return r;
+                        }
```

Comment:
The `bluechictl status <node> <unit>` is only for one unit, so no need to subscribe to a list of units. The code becomes a lot smaller then as well:
```suggestion
                        _cleanup_sd_bus_message_ sd_bus_message *m = NULL;
                        r = sd_bus_message_new_method_call(
                                        client->api_bus,
                                        &m,
                                        BC_INTERFACE_BASE_NAME,
                                        monitor_path,
                                        MONITOR_INTERFACE,
                                        "Subscribe");
                        if (r < 0) {
                                fprintf(stderr, "Failed creating subscription call: %s\n", strerror(-r));
                                return r;
                        }

                        sd_bus_message_append(m, "ss", node_name, unit_name);
```

Reply:
OK But is the for loop above to get multiple unit statuses?

Reply:
As we discussed, there is currently a bug in `bluechictl status <node> <unit1> <unit2>...` - I created issue https://github.com/eclipse-bluechi/bluechi/issues/869. 
@ArtiomDivak Could you fix this (and maybe include the other memory leak) first and then continue on this PR?